ludwig/automl/base_config.py

```diff
@@ -133,17 +141,48 @@ def _create_default_config(
     )
 
     model_configs = {}
-    for model_name, path_to_defaults in model_defaults.items():
-        default_model_config = load_yaml(path_to_defaults)
-        default_model_config.update(input_and_output_feature_config)
-        default_model_config['hyperopt']['executor'].update(
-            experiment_resources)
-        default_model_config['hyperopt']['executor']['time_budget_s'] = time_limit_s
-        model_configs[model_name] = default_model_config
+
+    # read in base config and update with experiment resources
+    base_automl_config = load_yaml(BASE_AUTOML_CONFIG)
+    base_automl_config["hyperopt"]["executor"].update(
+        experiment_resources
+    )
+    base_automl_config["hyperopt"]["executor"][
+        "time_budget_s"
+    ] = time_limit_s
+    if time_limit_s is not None:
+        num_samples = base_automl_config["hyperopt"]["sampler"]["num_samples"]
+        if num_samples is not None:
+            # allow trials to get 2x even division, since some trials perform better than avg
+            base_automl_config["hyperopt"]["sampler"]["scheduler"][
```

Comment:
I'm curious what happens if we let the per trial max_t equal the total training time. Would this result in some of the `num_samples` never getting tried? I would hope that hyperband would be fair in its scheduling so that it would try all the samples for a small amount of time before letting the best trials run indefinitely.

Reply:
Well, I thought I tried this and observed starvation of trials, but let me try it again to double-check.

Reply:
Okay, I just observed that for my Ray Tune async hyperband run, if specify num_samples=20
and set max_t == time_limit_s, then the first 3 trials are run for max_t time and no other trials are run.

Reply:
Thanks for confirming, sounds like this is the right thing to do for now, then. Would be great if we could find a way to both explore all trials and let trials run as long as they can.

Reply:
just posted this question to the ray tune slack channel